src/main/java/cs2103/duke/Deadline.java

```diff
@@ -7,12 +7,13 @@
  * This class encapsulates a Deadline Task object, which inherits from the Task class.
  */
 public class Deadline extends Task {
-
+    protected int index;
```

Comment:
I think it is better to use less general names and more specific names for better understanding by other readers.
For example, you can change the name here to indicate the context of the index.